modules/terminations.stanza

```diff
@@ -28,6 +28,43 @@ public pcb-enum ocdb/terminations/LVDSTerminationMethod :
   LVDSTerminationACDualResistorSplitCap
   LVDSTerminationACDualResistorSplitReduced
 
+public pcb-enum ocdb/terminations/USB-C-ConfigurationChannel :
+  SourceDefault
+  Source5V-1A5
+  Source5V3A0
+  SinkDefault
+
+; Terminates USB-C based on pull voltage and USB-C configuration
+public defn terminate-usb-c (usb-c-bundle:JITXObject, pull-up-net:JITXObject, pull-up-voltage:Double, method:USB-C-ConfigurationChannel) :
+  if port-type(usb-c-bundle) != usb-c :
+    fatal("Bundle type passed into terminate-usb-c must be bundle type usb-c.")
+  var termination-resistor-value:Double = 5.1e3 ; Default Sink termination value
+  inside pcb-module:
+    switch(method) :
+      SourceDefault:
+        termination-resistor-value = closest-std-val((pull-up-voltage / 80.0e-6) - 5.1e3, 5.0)
+        public inst term : chip-resistor(termination-resistor-value)[2]
+        net (usb-c-bundle.vbus.vdd term[0].p[2] term[1].p[2])
+        net (term[0].p[1] usb-c-bundle.cc[1])
+        net (term[1].p[1] usb-c-bundle.cc[2])
+      Source5V-1A5:
+        termination-resistor-value = closest-std-val((pull-up-voltage / 180.0e-6) - 5.1e3, 5.0)
+        public inst term : chip-resistor(termination-resistor-value)[2]
+        net (usb-c-bundle.vbus.vdd term[0].p[2] term[1].p[2])
+        net (term[0].p[1] usb-c-bundle.cc[1])
+        net (term[1].p[1] usb-c-bundle.cc[2])
+      Source5V3A0:
+        termination-resistor-value = closest-std-val((pull-up-voltage / 330.0e-6) - 5.1e3, 5.0)
+        public inst term : chip-resistor(termination-resistor-value)[2]
+        net (usb-c-bundle.vbus.vdd term[0].p[2] term[1].p[2])
+        net (term[0].p[1] usb-c-bundle.cc[1])
+        net (term[1].p[1] usb-c-bundle.cc[2])
+      SinkDefault:
+        public inst term : chip-resistor(termination-resistor-value)[2]
+        net (usb-c-bundle.vbus.gnd term[0].p[2] term[1].p[2])
+        net (term[0].p[1] usb-c-bundle.cc[1])
+        net (term[1].p[1] usb-c-bundle.cc[2])
+
```

Comment:
In your equation for the term resistor value, do you have a guarantee that pull-up-voltage/[80e-6, 180e-6, 330e-6] is always greater than 5.1e3? Otherwise the argument to chip-resistor() can be negative.

Reply:
In addition, you should refactor your logic to remove repeated statements:
```
  inside pcb-module:
    val term-current = 
      switch(method) :
        SourceDefault:
          80.0e-6
        Source5V-1A5:
          180.0e-6
        Source5V3A0:
          330.0e-6
        SinkDefault:
          0.0

    val termination-resistor-value = 
      if term-current != 0.0 :
        closest-std-val((pull-up-voltage / term-current) - 5.1e3, 5.0)
      else :
        5.1e3 ; Default Sink termination value
    public inst term : chip-resistor(termination-resistor-value)[2]
    net (usb-c-bundle.vbus.vdd term[0].p[2] term[1].p[2])
    net (term[0].p[1] usb-c-bundle.cc[1])
    net (term[1].p[1] usb-c-bundle.cc[2])
```

Reply:
If you use pull-up-voltage as Toleranced, then the access to that should be (depending on the limit analysis):
```
typ(pull-up-voltage) 
or 
min(pull-up-voltage)
or
max(pull-up-voltage)
```

Reply:
the netting is different. I would have to add two sets of switch statements to do this

Reply:
I think that you can do it with one switch statement returning the val of current and net as a tuple. I missed the gnd net used in the SinkDefault, but the same structure can be used.